### PR TITLE
add golden tests for CupertinoDatePicker

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
## Description

Add golden tests for `CupertinoDatePicker`
dependent on golden files: https://github.com/flutter/goldens/pull/29

## Related Issues

#30801

## Tests

I added the following tests:

- DatePicker golden tests

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/test/cupertino/date_picker_test.dart [hunk 1]
```diff
@@ -811,6 +811,35 @@ void main() {
...
+      );
+
+      // slightly drag the hour component to make the current hour off-center.
+      await tester.drag(find.text('4'), Offset(0, _kRowOffset.dy/2));
+      await tester.pump();
+
+      await expectLater(
+        find.byType(CupertinoDatePicker),
+        matchesGoldenFile('date_picker_test.datetime.drag.png')
+      );
```
Comment (dnfield) [inline packages/flutter/test/cupertino/date_picker_test.dart]: We should skip this on non-Linux platforms

File: packages/flutter/test/cupertino/date_picker_test.dart [hunk 2]
```diff
@@ -811,6 +811,35 @@ void main() {
...
+              onDateTimeChanged: (_) {},
+            )
+          )
+        )
+      );
+
+      await expectLater(
+        find.byType(CupertinoDatePicker),
+        matchesGoldenFile('date_picker_test.datetime.initial.png')
+      );
```
Comment (dnfield) [inline packages/flutter/test/cupertino/date_picker_test.dart]: Skip if !Platform.isLinux

**Comment by dnfield** [comment]
@fkorotkov - The git errors on these windows builds seem strange - looks like Git isn't configured properly on Windows instances right now.

**Comment by fkorotkov** [comment]
Nothing has changed to the images in the last weeks https://hub.docker.com/r/cirrusci/android-sdk/tags and it seems it's passing for master 🤔

**Comment by fkorotkov** [comment]
I've added https://github.com/cirruslabs/docker-images-android/pull/10 which should help with the issue but I still wonder why it started happening suddenly. New images will be available in ~30 minutes so you can retry.

File: packages/flutter/test/cupertino/date_picker_test.dart [hunk 3]
```diff
@@ -811,6 +813,37 @@ void main() {
...
+        )
+      );
+
+      await expectLater(
+        find.byType(CupertinoDatePicker),
+        matchesGoldenFile('date_picker_test.datetime.initial.png'),
+        skip: !Platform.isLinux
+      );
+
+      // slightly drag the hour component to make the current hour off-center.
```
Comment (xster) [inline packages/flutter/test/cupertino/date_picker_test.dart]: // S

Capitalize

**Review by xster — APPROVED** [review]
LGTM, thanks

File: packages/flutter/test/cupertino/date_picker_test.dart [hunk 4]
```diff
@@ -811,6 +813,37 @@ void main() {
...
+      await tester.drag(find.text('4'), Offset(0, _kRowOffset.dy/2));
```
Comment (xster) [inline packages/flutter/test/cupertino/date_picker_test.dart]: super nit: `_kRowOffset.dy / 2`

**Pull request merged**